cli/azd/pkg/azsdk/funcapp_host_client.go

```diff
@@ -117,9 +117,12 @@ func (c *FuncAppHostClient) Publish(
 
 func (c *FuncAppHostClient) waitForDeployment(ctx context.Context, location string) (PublishResponse, error) {
 	// This frequency is recommended by the service team.
-	polLDelay := 1 * time.Second
+	pollDelay := 1 * time.Second
```

Comment:
Have we considered using the `go-retry` package that we have used in the past? It seems to already support the polling logic with max retries, etc.

Reply:
My preference would be:

1. Consider azure-sdk-for-go retry policies
2. If implementing HTTP directly, consider handling error conditions directly.

My main hesitancies against `go-retry` is that 1. It abstracts away for-loops, which "hides" the true performance cost from the programmer, and 2. It makes it too easy to add inner retries and outer retries that lead to exponential retry loops.

This is primarily speaking from my own experience on previous teams, where a client tool ended up retrying for an hour, because we added multiple retries at different layers. They all looked fine going through code review -- but it's only when you stare at a for-loop long enough that we understand the cost of execution that we are imposing on the program.